src/code/TwoHeadArena.c

```diff
@@ -103,7 +103,7 @@ void* THA_AllocTailAlign(TwoHeadArena* tha, size_t size, uintptr_t mask) {
  * @return Remaining size. A negative number indicates an overflow.
  */
 s32 THA_GetRemaining(TwoHeadArena* tha) {
-    return (s32)((u8*)tha->tail - (u8*)tha->head);
+    return (uintptr_t)tha->tail - (uintptr_t)tha->head;
```

Comment:
Actually won't this output an unsigned thing that then gets cast? I think keeping it as u8* is a much better option.

Reply:
We can't keep this as `u8*` without casting somewhere, that's the point of `-Wint-conversion`

Reply:
So cast to the return type afterwards.